src/server/scripts/Events/fireworks_show.cpp

```diff
@@ -0,0 +1,903 @@
+        {
+            _events.Update(diff);
+
+            time_t time = sWorld->GetGameTime();
+            tm localTm;
+            localtime_r(&time, &localTm);
+
+            // Start
+            if (((localTm.tm_min == 0 && localTm.tm_sec == 0) && !started && IsHolidayActive(HOLIDAY_FIREWORKS_SPECTACULAR)) ||
+                ((localTm.tm_hour == 0 && localTm.tm_min == 0 && localTm.tm_sec == 0) && !started && IsEventActive(GAME_EVENT_NEW_YEAR)))
+            {
+                _events.ScheduleEvent(EVENT_CHEER, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                started = true;
+            }
+
+            // Event is active
+            if ((localTm.tm_min >= 0 && localTm.tm_sec >= 1 && localTm.tm_min <= 9 && localTm.tm_sec <= 59) && !started && IsHolidayActive(HOLIDAY_FIREWORKS_SPECTACULAR))
+            {
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                started = true;
+            }
+
+            // Stop
+            if ((localTm.tm_min == 10 && localTm.tm_sec == 0) || (localTm.tm_min == 10 && localTm.tm_sec == 0 && localTm.tm_hour == 0 && started == true))
+            {
+                started = false;
+                _events.ScheduleEvent(EVENT_CHEER, Seconds(1));
+                _events.CancelEvent(EVENT_FIRE);
+            }
+
+            // New Year (Only!) - One more big bang!
+            if ((localTm.tm_min == 10 && localTm.tm_sec == 30 && localTm.tm_hour == 0) && IsEventActive(GAME_EVENT_NEW_YEAR) && big == true)
+            {
+                big = false;
+                _events.ScheduleEvent(EVENT_CHEER, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+            }
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_CHEER:
+                    {
+                        go->PlayDistanceSound(CheerPicker());
+                        break;
+                    }
+                    case EVENT_FIRE:
+                    {
+                        auto it = pos.find(go->GetZoneId());
+                        if (it != pos.end())
+                        {
+                            std::vector<Position> const& zoneFireworks = it->second;
+                            uint32 rndpos = urand(0, zoneFireworks.size());
```

Comment:
that could cause overflows if rndpos is zoneFireworks.size(), you can use `Position const& randomPos = Trinity::Containers::SelectRandomContainerElement(it->second)`